clarifai/client/auth/stub.py

```diff
@@ -20,6 +21,34 @@
 _threadpool = ThreadPoolExecutor(100)
 
 
+def validate_response(response, attempt, max_attempts):
+  # Helper function to handle simple response validation
+  def handle_simple_response(response):
+    if hasattr(response, 'status') and hasattr(response.status, 'code'):
+      if (response.status.code in throttle_status_codes) and attempt < max_attempts:
+        logging.debug('Retrying with status %s' % str(response.status))
+        return None  # Indicates a retry is needed
+      else:
+        return response
+
+  # Check if the response is an instance of a gRPC streaming call
+  if isinstance(response, grpc._channel._MultiThreadedRendezvous):
+    try:
+      # Check just the first response in the stream for validation
+      first_res = next(response)
+      validated_response = handle_simple_response(first_res)
+      if validated_response is not None:
+        # Have to return that first response and the rest of the stream.
+        return itertools.chain([validated_response], response)
```

Comment:
@luv-bansal see here where I had to change the logic so that it includes the first chunk off the stream. Tested that this works and will merge now.

Reply:
Yeah, this seems good.